packages/router/src/grpc/view-protocol-server/broadcast-transaction.ts

```diff
@@ -15,15 +19,38 @@ export const broadcastTransaction: Impl['broadcastTransaction'] = async (req, ct
   // start subscription early to prevent race condition
   const subscription = indexedDb.subscribe('TRANSACTION_INFO');
 
-  const id = await tendermint.broadcastTx(req.transaction);
+  const id = new TransactionId({ inner: await sha256Hash(req.transaction.toBinary()) });
+
+  const broadcastId = await tendermint.broadcastTx(req.transaction);
+  if (!id.equals(broadcastId)) {
+    console.error('broadcast transaction id disagrees', id, broadcastId);
```

Comment:
```suggestion
```

Reply:
no. this is important because if for some reason conversion to string fails, it's still logged